src/backend/distributed/commands/multi_copy.c

```diff
@@ -3187,26 +3190,47 @@ CheckCopyPermissions(CopyStmt *copyStatement)
 	rel = table_openrv(copyStatement->relation,
 	                  is_from ? RowExclusiveLock : AccessShareLock);
 
-	range_table = CreateRangeTable(rel, required_access);
+	range_table = CreateRangeTable(rel);
 	RangeTblEntry *rte = (RangeTblEntry*) linitial(range_table);
 	tupDesc = RelationGetDescr(rel);
 
+#if PG_VERSION_NUM >= PG_VERSION_16
+	/* create permission info for rte */
+	RTEPermissionInfo *perminfo = GetFilledPermissionInfo(rel->rd_id, rte->inh, required_access);
+#else
+	rte->requiredPerms = required_access;
```

Comment:
Is it possible to move this into the above routine, and the only line that executes for pre-16. That way we can remove the version macro here? You can ignore, if you feel like it's easy here

Reply:
I couldn't quite understand what you mean. I wonder if you can write a suggestion box.

Reply:
Something on these lines
![image](https://github.com/citusdata/citus/assets/44680808/43620add-9906-4d56-bdbf-c3f6a4df38c4)

Reply:
If you use the method, you did for other routines in `pg_version_compat.h` , then we will have two versions of the same routine. If you think, it's cumbersome, please ignore my comment.

Reply:
I prefer it as it is, because pre-pg16 if there was something like FilledPermissionInfo it wouldn't have just requiredPerms.
The code may not be very beautiful right now, but there will be other PermissionInfo changes in the bigger PG16 PR, and this stays in line with them.